### PR TITLE
Removing deprecated DefinitionLocation class

### PR DESCRIPTION
**Pull request by mamazu**
And replacing it with the more generally named `Location` class.

### CODE REVIEW

File: lib/Extension/LanguageServerReferenceFinder/Tests/Unit/Handler/GotoDefinitionHandlerTest.php [hunk 1]
```diff
@@ -31,10 +29,11 @@ class GotoDefinitionHandlerTest extends TestCase
     public function testGoesToDefinition(): void
     {
         $document = TextDocumentBuilder::create(self::EXAMPLE_TEXT)->uri(self::EXAMPLE_URI)->build();
+
         $locations = [
             new TypeLocation(
                 TypeFactory::class('Foo'),
-                new DefinitionLocation($document->uri(), ByteOffset::fromInt(2))
+                new PhpactorLocation($document->uriOrThrow(), ByteOffset::fromInt(2))
```
Comment (dantleech) [inline lib/Extension/LanguageServerReferenceFinder/Tests/Unit/Handler/GotoDefinitionHandlerTest.php:36]: can we remove the alias and just have `Location` ?

File: lib/ReferenceFinder/Tests/Unit/LocationTest.php [hunk 2]
```diff
@@ -3,18 +3,18 @@
 namespace Phpactor\ReferenceFinder\Tests\Unit;
 
 use PHPUnit\Framework\TestCase;
-use Phpactor\ReferenceFinder\DefinitionLocation;
 use Phpactor\TextDocument\ByteOffset;
+use Phpactor\TextDocument\Location;
 use Phpactor\TextDocument\TextDocumentUri;
 
-class DefinitionLocationTest extends TestCase
+class LocationTest extends TestCase
```
Comment (dantleech) [inline lib/ReferenceFinder/Tests/Unit/LocationTest.php:10]: this test no longer maps to a class, we can either simply remove it or move the test cases to the test for `TextDocument\Location`

**Review by dantleech — APPROVED** [review]

**Comment by dantleech** [comment]
:+1:  looks good!

File: lib/Extension/LanguageServerReferenceFinder/Tests/Unit/Handler/GotoDefinitionHandlerTest.php (hunk 1 above)
Comment (mamazu) [reply lib/Extension/LanguageServerReferenceFinder/Tests/Unit/Handler/GotoDefinitionHandlerTest.php:36]: In this file there is this `Phpactor\TextDocument\Location` and `Phpactor\LanguageServerProtocol\Location` the later one is not aliased so the first one needs an alias.

File: lib/ReferenceFinder/Tests/Unit/LocationTest.php (hunk 2 above)
Comment (mamazu) [reply lib/ReferenceFinder/Tests/Unit/LocationTest.php]: Removing since there already is a test for the location in the TextDocument namespace.

**Comment by dantleech** [comment]
Danke!

**Pull request merged**